### PR TITLE
fix: Encode `nan`, `inf` and `-inf` values as `null` in Singer stream

### LINKED ISSUE #2213: bug: ValueError: Out of range float values are not JSON compliant
### Singer SDK Version

0.34.1

### Is this a regression?

- [ ] Yes

### Python Version

3.9

### Bug scope

Taps (catalog, state, etc.)

### Operating System

MacOS

### Description

I get `ValueError: Out of range float values are not JSON compliant`  when trying to sync Amazon Document DB collection. 

The probably reason is that `simple_json.dumps()` has `allow_nan=False` default, while standard `dumps()` function from standard `json` library has `allow_nan=True`. 

As there is no way to pass configuration from outside could we add this functionality? I can introduce relevant fix. 

### Code

_No response_

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Changes the behavior of JSON serialization for Singer output to convert `nan`, `inf` and `-inf` values to `null` in compliance with the ECMA-262 specification.

## Implementation details

- `simplejson` supports this with the `ignore_nan=True`[^1] parameter.
- `msgspec` already does this by default[^2].

## Related

- Closes #2213

[^1]: https://simplejson.readthedocs.io/en/latest/#simplejson.dumps
[^2]: https://jcristharif.com/msgspec/supported-types.html#float

## Summary by Sourcery

Modify JSON serialization in Singer output to convert special float values (`nan`, `inf`, `-inf`) to `null`

Bug Fixes:
- Ensure proper handling of special float values in JSON serialization

Tests:
- Add test case to verify `nan` values are encoded as `null`

----
📚 Documentation preview 📚: https://meltano-sdk--3019.org.readthedocs.build/en/3019/

### CODE REVIEW

File: tests/singerlib/encoding/test_simple.py [hunk 1]
```diff
@@ -104,3 +104,17 @@ def test_write_message():
     assert out.getvalue() == (
         '{"type":"RECORD","stream":"test","record":{"id":1,"name":"test"}}\n'
     )
```
[the remark this answers is not in the document]
Comment (edgarrmondragon) [reply tests/singerlib/encoding/test_simple.py:106]: Can you make a code suggestion?
